source/usage-examples/watch.txt

```diff
@@ -0,0 +1,139 @@
+The ``watch()`` method optionally takes an **aggregation pipeline**  which
+consists of an array of **stages** as the first parameter to filter and
+transform the change event output as follows:. 
+
+.. code-block:: java
+
+   List<Bson> pipeline = Arrays.asList(
+                           Aggregates.match(
+                              Filters.lt("fullDocument.runtime", 15)));
+   ChangeStreamIterable<Document> changeStream = database.watch(pipeline);
+
+To configure options for the change stream, use member methods of the
+``ChangeStreamIterable`` object returned by ``watch()``. Refer to the
+links at the end of this section for more information on the settings
+you can configure.
+
+The ``watch()`` method returns an instance of a ``ChangeStreamIterable``.
+You can call ``forEach()`` on the ``ChangeStreamIterable`` to handle
+events as they occur, or you can use the ``iterator()`` method to access
+the underlying ``MongoIterator``. This allows you to call manual methods
+on the change stream such as ``hasNext()`` to check for remaining
+documents in the stream and ``next()`` to request the next document in
+the stream. See the link to the ``ChangeStreamIterable`` API
+documentation below for more details on the available methods.
+
+Visit the following resources for additional material on the classes and
+methods presented above:
+
+- :manual:`change streams </changeStreams/>`
+- :manual:`change events </reference/change-events/>`
+- :manual:`aggregation pipeline </reference/operator/aggregation-pipeline/>`
+- :manual:`aggregation stages </changeStreams/#modify-change-stream-output>`
+- :java-sync-api:`ChangeStreamIterable <com/mongodb/client/ChangeStreamIterable.html>`
+- :java-sync-api:`MongoCollection.watch() <com/mongodb/client/MongoCollection.html#watch()>`
+- :java-sync-api:`MongoDatabase.watch() <com/mongodb/client/MongoDatabase.html#watch()>`
+- :java-sync-api:`MongoClient.watch() <com/mongodb/client/MongoClient.html#watch()>`
+
+Process the Change Stream Events
+--------------------------------
+
+To capture events from a change stream, call the ``forEach()`` method
+with a callback function as shown below:
+
+.. code-block:: java
+
+   changeStream.forEach(event -> System.out.println("Change observed: " + event));
+
+The callback function triggers when a change event is emitted. You can
+specify logic in the callback to process the event document when it is
+received.
+
+.. admonition:: forEach() blocks the current thread
+   :class: important
+
+   Calls to ``forEach()`` block the current thread as long as the
+   corresponding change stream listens for events. If your program
+   needs to continue executing other logic, such as processing requests or
+   responding to user input, consider creating and listening to your
+   change stream in a separate thread.
+
+.. note::
+
+   For update operation change events, change streams only return the modified
+   fields by default rather than the full updated document. You can configure
```

Comment:
Nitpick: how about "complete" instead of "full"? It's a little confusing whether you mean "complete" or "fully-updated" with that word combo.